src/fits.rs

```diff
@@ -236,10 +225,22 @@ where
 
         // Data block
 
-        Ok(Self { header })
+        Ok(Self { header, data_unit_byte_offset: 0 }) /* Set in impl Iterator for Fits */
```

Comment:
compute the value of data_unit_byte_offset here by calling stream_position just before returning the HDU.
Then in Fits::next you can set the pos_start_cur_du with the one from the hdu.